packages/web-client/app/models/workflow/workflow-card.ts

```diff
@@ -25,14 +37,26 @@ export class WorkflowCard extends WorkflowPostable {
         this.isComplete = false;
       }
     };
+    this.check = options.check;
   }
   get session(): WorkflowSession | undefined {
     return this.workflow?.session;
   }
 
   @action onComplete() {
-    this.workflow?.emit('visible-postables-changed');
-    this.isComplete = true;
+    if (this.check) {
+      this.check().then((checkResult) => {
+        if (checkResult.success) {
+          this.workflow?.emit('visible-postables-changed');
```

Comment:
I know this is not *new* in your PR, but.... why do we have to do this? Isn't setting isComplete enough state for us to react to?

Reply:
'visible-postables-changed' is for test waiters/ the animated workflow.

Reply:
Renamed to `visible-postables-will-change` and added a comment about why we are emitting this.